controllers/appmesh/cloudmap_controller.go

```diff
@@ -129,58 +72,35 @@ func (r *cloudMapReconciler) reconcile(req ctrl.Request) error {
 	}
 
 	if !vNode.DeletionTimestamp.IsZero() {
-		return r.deleteCloudMapResources(ctx, vNode)
+		return r.cleanupCloudMapResources(ctx, vNode)
 	}
 	return r.reconcileVirtualNodeWithCloudMap(ctx, vNode)
 }
 
 func (r *cloudMapReconciler) reconcileVirtualNodeWithCloudMap(ctx context.Context, vNode *appmesh.VirtualNode) error {
 
-	if !r.isCloudMapEnabledForVirtualNode(vNode) {
+	if !cloudmap.IsCloudMapEnabledForVirtualNode(vNode) {
 		r.log.V(1).Info("CloudMap not enabled for", " virtualNode: %s", vNode.Name)
 		return nil
 	}
-
-	if !r.isPodSelectorDefinedForVirtualNode(vNode) {
+	if !cloudmap.IsPodSelectorDefinedForVirtualNode(vNode) {
```

Comment:
ideally, if podSelector is not defined, we should still create the cloudMap service if not exists.

But we don't manage the instances in the cloudMap service. (so that it's possible for users to populate the cloudmap service instances by other means :D)